src/generate/writer.rs

```diff
@@ -2,25 +2,31 @@ use std::{fs::File, io::Write};
 
 pub struct CppWriter {
     pub stream: File,
-    pub indent: u16
+    pub indent: u16,
+    pub newline: bool
 }
 
 impl CppWriter {
     pub fn indent(&mut self) {
-        self.indent += 4;
+        self.indent += 1;
     }
     pub fn dedent(&mut self) {
-        if self.indent < 4 {
+        if self.indent == 0 {
             panic!("Indent should not be dedented when already less than 4!")
         }
-        self.indent -= 4;
+        self.indent -= 1;
     }
 }
 
 impl Write for CppWriter {
     fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
         // TODO: One day we will write indented
+        if self.indent > 0 && self.newline {
+            self.stream.write_all("\t".repeat(self.indent.into()).as_bytes())?;

```

Comment:
:( Also, we don't need to use indented writers because we'll let clang-format deal with it for us, so we can emit in whatever format we care about honestly.

Reply:
bruh